### PR TITLE
apps: Add a flag to disable SBOM generation

### PR DESCRIPTION
**Pull request by doanac**

### CODE REVIEW

**Comment by doanac** [comment]
@vanmaegima - this seems to work in my factory. You think this will work for the customer request?

File: apps/build.sh [hunk 1]
```diff
@@ -175,10 +175,14 @@ for x in $IMAGES ; do
 
 	echo "Build step $((completed+2)) of $total is complete"
 
-	status "Doing a syft SBOM scan"
-	sbom_dst=/archive/sboms/${ct_base}/${ARCH}.spdx.json
-	mkdir -p $(dirname $sbom_dst)
-	syft ${ct_base}:$TAG-$ARCH -o spdx-json > $sbom_dst
+	if [ -z "$DISABLE_SBOM" ] ; then
```
Comment (mike-scott) [inline apps/build.sh:178]: @doanac To implement this a user adds this to the `params` block of `containers`?
```
DISABLE_SBOM: "1"
```

Comment (doanac) [reply apps/build.sh:178]: yep. i'm testing that now

Comment (doanac) [reply apps/build.sh:178]: here it is disabled - https://app.foundries.io/factories/andy-corp/targets/77/ 
here it is enabled(default) - https://app.foundries.io/factories/andy-corp/targets/78/

**Comment by vanmaegima** [comment]
@doanac If that is part of the `containers` block, would that impact platform builds as well? I remember preloaded apps inherit the block from containers, so it'd be good to understand the impact there. I believe the request here is to disable SBOMs for containers only, but I'll double check.

**Comment by quaresmajose** [comment]
> This only disables for containers. SBOM generation for Platform is done via Yocto - I'm not sure how you disable that.

The yocto SPDX can be also be disabled but I don't see much benefit in doing it, in the kirkstone branch it also have negative side effects like reducing the sstate-cache reuse.

**Review by vanmaegima — APPROVED** [review]

**Review by MrCry0 — APPROVED** [review]

**Comment by doanac** [comment]
rebased on master

**Pull request merged**